qa/rpc-tests/zcpourdoublespend.py

```diff
@@ -86,50 +111,73 @@ def run_test(self):
                                           {zcaddress:(49.9*2)-0.1},
                                           0, 0.1)
 
-        # (a)    Node 1 will spend pour AB, then attempt to
+        # (a)    Node 0 will spend pour AB, then attempt to
         # double-spend it with BC. It should fail before and
-        # after Node 1 mines blocks.
+        # after Node 0 mines blocks.
         #
         # (b)    Then, Node 2 will spend BC, and mine 5 blocks.
         # Node 1 connects, and AB will be reorg'd from the chain.
         # Any attempts to spend AB or CD should fail for
         # both nodes.
         #
-        # (c)    Then, Node 1 will spend AD, which should work
+        # (c)    Then, Node 0 will spend AD, which should work
         # because the previous spend for A (AB) is considered
-        # invalid.
+        # invalid due to the reorg.
 
         # (a)
 
-        self.nodes[0].sendrawtransaction(pour_AB["rawtxn"])
+        AB_txid = self.nodes[0].sendrawtransaction(pour_AB["rawtxn"])
 
         self.expect_cannot_pour(self.nodes[0], pour_BC["rawtxn"])
 
+        # Wait until node[1] receives AB before we attempt to double-spend
+        # with BC.
+        print "Waiting for AB_txid...\n"
+        while True:
```

Comment:
Yes. I could remove the loop and use `sync_blocks` which I suspect does something similar, but this particular loop checks the mempool, which we don't want to neglect.